core/src/main/kotlin/org/evomaster/core/search/gene/NumericStringGene.kt

```diff
@@ -0,0 +1,114 @@
+package org.evomaster.core.search.gene
+
+import org.evomaster.core.output.OutputFormat
+import org.evomaster.core.search.impact.impactinfocollection.value.OptionalGeneImpact
+import org.evomaster.core.search.impact.impactinfocollection.value.numeric.NumericStringGeneImpact
+import org.evomaster.core.search.service.AdaptiveParameterControl
+import org.evomaster.core.search.service.Randomness
+import org.evomaster.core.search.service.mutator.MutationWeightControl
+import org.evomaster.core.search.service.mutator.genemutation.AdditionalGeneMutationInfo
+import org.evomaster.core.search.service.mutator.genemutation.SubsetGeneSelectionStrategy
+import java.math.BigDecimal
+
+class NumericStringGene(
+    name: String,
+    /**
+     * inclusive
+     *
+     * note that precision would represent maxLength
+     */
+    val minLength: Int,
```

Comment:
if this `minLength` constraints is needed, shouldn't it be inside `BigDecimalGene`? otherwise, how can we ensure a mutation on this latter will not break the constraint?

Reply:
For BigDecimalGene, there is no lower bound for the digits. 
I add a TODO for this String if it exists.
It is a bit tricky.
It depends on how the string would be further dealt with, eg, we could fill 0 at left. 
or we need to control min and max.